atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueue.java

```diff
@@ -161,56 +168,67 @@ public int getNumShards() {
             this.progress = progress;
             this.numShards = numShards;
             this.cells = cells;
             this.timestamps = timestamps;
             this.metrics = metrics;
             this.kvs = kvs;
             this.timelock = timelock;
+            this.runtimeConfigSupplier = runtimeConfigSupplier;
         }
 
         static SweepQueueFactory create(
                 TargetedSweepMetrics metrics,
                 KeyValueService kvs,
                 TimelockService timelock,
-                Supplier<Integer> shardsConfig) {
+                Supplier<Integer> shardsConfig,
+                Supplier<TargetedSweepRuntimeConfig> runtimeConfigSupplier) {
             // It is OK that the transaction service is different from the one used by the transaction manager,
             // as transaction services must not hold any local state in them that would affect correctness.
             TransactionService transaction = TransactionServices.createRaw(
                     kvs,
                     new TimelockTimestampServiceAdapter(timelock),
                     false);
-            return create(metrics, kvs, timelock, shardsConfig, transaction);
+            return create(metrics, kvs, timelock, shardsConfig, transaction, runtimeConfigSupplier);
         }
 
         static SweepQueueFactory create(
                 TargetedSweepMetrics metrics,
                 KeyValueService kvs,
                 TimelockService timelock,
                 Supplier<Integer> shardsConfig,
-                TransactionService transaction) {
+                TransactionService transaction,
+                Supplier<TargetedSweepRuntimeConfig> runtimeConfigSupplier) {
             Schemas.createTablesAndIndexes(TargetedSweepSchema.INSTANCE.getLatestSchema(), kvs);
             ShardProgress shardProgress = new ShardProgress(kvs);
             Supplier<Integer> shards = createProgressUpdatingSupplier(shardsConfig, shardProgress,
                     SweepQueueUtils.REFRESH_TIME);
             WriteInfoPartitioner partitioner = new WriteInfoPartitioner(kvs, shards);
             SweepableCells cells = new SweepableCells(kvs, partitioner, metrics, transaction);
             SweepableTimestamps timestamps = new SweepableTimestamps(kvs, partitioner);
-            return new SweepQueueFactory(shardProgress, shards, cells, timestamps, metrics, kvs, timelock);
+            return new SweepQueueFactory(
+                    shardProgress, shards, cells, timestamps, metrics, kvs, timelock, runtimeConfigSupplier);
         }
 
         private SweepQueueWriter createWriter() {
             return new SweepQueueWriter(timestamps, cells);
         }
 
         private SweepQueueReader createReader() {
-            return new SweepQueueReader(timestamps, cells);
+            return new SweepQueueReader(
+                    timestamps, cells, () -> runtimeConfigSupplier.get().maximumPartitionsToBatchInSingleRead());
```

Comment:
I'm not sure if pushing the whole RuntimeConfig all the way down just for this one value makes sense opposed to just having a Supplier<Integer> and leaking less abstraction. On the other hand, not a fan of injecting magic numbers out of thin air either, so feel like there is no great option here.

Reply:
Yep. The alternative is supplying the number, but I didn't think it was too bad since at least the pieces engaged here are clearly relating to Targeted sweep.